core/src/main/scala/algebra/ring/Ring.scala

```diff
@@ -32,11 +32,11 @@ trait Ring[@sp(Int, Long, Float, Double) A] extends Any with Rig[A] with Rng[A]
   def fromInt(n: Int): A = sumN(one, n)
 }
 
-trait RingFunctions {
+trait RingFunctions extends AdditiveGroupFunctions with MultiplicativeMonoidFunctions {
```

Comment:
Scala should prioritize the most "specific" trait. So, if both a Ring[A] and a Field[A] are in scope, and we ask for an implicit Ring[A], then the field *should* be chosen. However, relying on this rule is pretty hairy (IMO) and I think we should parameterize.